telegram/bot.py

```diff
@@ -208,6 +208,12 @@ def __init__(
                 private_key, password=private_key_password, backend=default_backend()
             )
 
+    def __setattr__(self, key: str, value: object) -> None:
+        if issubclass(self.__class__, Bot) and self.__class__.__name__ != 'Bot':
```

Comment:
shouldn't `self.__class__ is Bot` suffice and also be safer? the custom subclasses name could also be `Bot`.
Same for the other places where we use this logic.

Reply:
It's more pythonic and safer but even that would fail if the subclass' name is `Bot`, but the user shouldn't be doing that anyway..